src/cscapi/client.py

```diff
@@ -153,55 +153,79 @@ def _send_signals_by_machine_id(
-                machine_to_process = self._prepare_machine(machine_to_process)
+                try:
+                    machine_to_process = self._prepare_machine(machine_to_process)
+                except httpx.HTTPStatusError as exc:
+                    self.logger.error(
+                        f"Error while preparing machine {machine_to_process.machine_id}: {exc}"
+                    )
+                    machine_to_process.token = None
+                    retry_machines_to_process_attempts.append(machine_to_process)
+                    continue
                 if machine_to_process.is_failing:
                     self.logger.error(
-                        f"skipping sending signals for machine {machine_to_process.machine_id} as it's marked as failing"
+                        f"skipping sending signals for machine {machine_to_process.machine_id}"
+                        f" as it's marked as failing"
                     )
                     continue
 
-                self.logger.info(
-                    f"sending signals for machine {machine_to_process.machine_id}"
-                )
                 sent_signal_ids = []
                 try:
-                    sent_signal_ids = self._send_signals_to_capi(
-                        machine_to_process.token,
-                        signals_by_machineid[machine_to_process.machine_id],
+                    self.logger.info(
+                        f"sending signals for machine {machine_to_process.machine_id}"
                     )
-                    sent_signal_ids_count = len(sent_signal_ids)
-                    total_sent += sent_signal_ids_count
-                    self.logger.info(f"sent {sent_signal_ids_count} signals")
+                    if machine_to_process.token is not None:
+                        sent_signal_ids = self._send_signals_to_capi(
+                            machine_to_process.token,
+                            signals_by_machineid[machine_to_process.machine_id],
+                        )
+                        sent_signal_ids_count = len(sent_signal_ids)
+                        total_sent += sent_signal_ids_count
+                        self.logger.info(f"sent {sent_signal_ids_count} signals")
+                    else:
+                        self.logger.error(
+                            f"skipping sending signals for machine {machine_to_process.machine_id}"
+                            f" as it has no token"
+                        )
+                        continue
                 except httpx.HTTPStatusError as exc:
                     self.logger.error(
                         f"error while sending signals: {exc} for machine {machine_to_process.machine_id}"
                     )
                     if exc.response.status_code == 401:
-                        if attempt_count >= self.max_retries:
-                            self.storage.update_or_create_machine(
```

Comment:
I delete this piece of code, because we already update the machine in line 165 (after PR, line 157 before PR).

@rr404 : am I missing something ?